processor/memorylimiterprocessor/config.go

```diff
@@ -51,9 +51,34 @@ type Config struct {
 	MemorySpikePercentage uint32 `mapstructure:"spike_limit_percentage"`
 }
 
-var _ component.Config = (*Config)(nil)
+var (
+	_ component.Config = (*Config)(nil)
+)
 
 // Validate checks if the processor configuration is valid
 func (cfg *Config) Validate() error {
+	if cfg.CheckInterval <= 0 {
+		return errCheckIntervalOutOfRange
+	}
+	switch {
```

Comment:
Should we error out also if the memory limit is set in two different ways (via MiB and percentage)?

Reply:
Perhaps, I was keeping the existing logic but moving it to the validate.

Reply:
Okay, that's fair, since it would be a breaking change we can just warn for now